### PR TITLE
Messages

### PR DESCRIPTION
**Pull request by mheir**

### CODE REVIEW

**Comment by reblace** [comment]
How hard would it be to set this up without the dependency on Kafka? We'd previously talked about refactoring it a little bit to put a provider on the back end to separate Kafka from the actual message passing so we could sub in a simple event emitter on the back end for local testing.

**Comment by mheir** [comment]
@reblace I thought about removing the kafka dependency, but wasn't sure if that was in the scope of getting it ported over or doing a larger refactor now.  It wouldn't be too difficult overall, but it's a bit more than a little change.  Right now the sockets are set up such that they extend KafkaSocket and are tightly coupled to subscribing to kafka events.  It'd be a decent deviation from how all the other sockets work at the moment.  If that's something we want to do now, I can look into it.

File: src/client/app/admin/admin.component.html [hunk 1]
```diff
@@ -4,6 +4,7 @@
 		<li role="presentation"><a [routerLink]="['/admin/users', {clearCachedFilter: true}]" routerLinkActive="active">Users</a></li>
 		<li role="presentation"><a [routerLink]="['/admin/cacheEntries', {clearCachedFilter: true}]" routerLinkActive="active">Cache Entries</a></li>
 		<li role="presentation"><a [routerLink]="['/admin/euas', {clearCachedFilter: true}]" routerLinkActive="active">End User Agreement</a></li>
+		<li role="presentation"><a [routerLink]="['/admin/messages', {clearCachedFilter: true}]">Messages</a></li>
```
Comment (ymao1) [inline src/client/app/admin/admin.component.html]: Does this need a routerLinkActive="active"?

File: src/client/app/core/socket.service.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	}
+
+	public on(eventName: string, callback: Function) {
+		this.socket.on(eventName, (event: any) => {
+			callback.apply(this.socket, [event]);
+		});
+	}
+
+	public emit(eventName: string) {
+		console.log(eventName);
```
Comment (ymao1) [inline src/client/app/core/socket.service.ts]: Remove console.log?

Comment (mheir) [reply src/client/app/core/socket.service.ts]: fixed

File: src/client/app/admin/admin.component.html (hunk 1 above)
Comment (mheir) [reply src/client/app/admin/admin.component.html]: done

**Review by ymao1 — COMMENTED** [review]
Creating a new MOTD works fine and the toaster pops up. When I click the edit button, it takes me back to the create page.

Also, is there a reason the search bar doesn't stretch across the full page? It looks like it's offset on the left as well. I realize it looks the same on FH, but it kind of looks funny.

Finally, another thing that looks like it's not working on FH either is that when all the columns are selected, the ID column is empty. This looks like it's working on unrulyfire.

File: src/client/app/messages/admin/create-message.component.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import { Component } from '@angular/core';
+import { ManageMessageComponent } from './manage-message.component';
+import { Router } from '@angular/router';
+import { MessageService } from '../message.service';
+import { Message, MessageType } from '../message.class';
+import { ConfigService } from '../../core/config.service';
+import { AlertService } from '../../shared/alert.service';
+
+@Component({
+	selector: 'admin-create-user',
```
Comment (ymao1) [inline src/client/app/messages/admin/create-message.component.ts]: should this be admin-create-message and admin-edit-message (below)

File: src/client/app/messages/admin/create-message.component.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+	initialize() {
+		this.title = 'Create Message';
+		this.subtitle = 'Provide the required information to create a new message';
+		this.okButtonText = 'Create';
+		this.navigateOnSuccess = '/admin/messages';
+		this.message = new Message();
+		this.message.type = MessageType.MOTD;
+	}
+
+	handleBypassAccessCheck() {
```
Comment (ymao1) [inline src/client/app/messages/admin/create-message.component.ts]: this looks like a hold over from users?

File: src/client/app/welcome/external-links.component.html [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<accordion [closeOthers]="true" >
+	<accordion-group class="accordion-heading" [isOpen]="link.isOpen" (click)="link.isOpen = !link.isOpen" *ngFor="let link of links">
+		<div accordion-heading>
+			<a target="_blank" href="{{ link.href }}" (click)="handleLinkClick($event)">{{ link.title }}</a>
+			<i class="pull-right glyphicon"
+			   [ngClass]="{'glyphicon-chevron-down': link.isOpen, 'glyphicon-chevron-right': !link.isOpen}"></i>
```
Comment (ymao1) [inline src/client/app/welcome/external-links.component.html]: use FA icons instead?

**Comment by mheir** [comment]
@ymao1 Ok, fixed everything you commented about thus far

**Comment by mheir** [comment]
I modified the sockets so they are no longer coupled tightly to kafka.  There is a new event based socket that uses node's EventEmitter.  See default.js for config options.

**Pull request merged**